libsql-server/src/replication/primary/replication_logger_wal.rs

```diff
@@ -252,3 +255,144 @@ impl libsql_sys::wal::Wal for ReplicationLoggerWal {
         self.inner.last_fame_index()
     }
 }
+
+impl ReplicationLoggerWal {
+    fn inject_replication_index(&mut self) -> Result<()> {
+        let data = &mut [0; LIBSQL_PAGE_SIZE as _];
+        // We retreive the freshest version of page 1. Either most recent page 1 is in the WAL, or
+        // it is in the main db file
```

Comment:
@MarinPostma that implies we only create the logger on a db that was previously opened an initialized. Reasonable assumption, but maybe worth documenting somewhere? Or alternatively we can error-out on an empty file instead of unwrap() and slap a clear error message there

Reply:
you're right this error should not be ignored. I fixed it here: https://github.com/tursodatabase/libsql/pull/761, and added a test that checkpoints an empty db